project-root/Assets/Scripts/NavMeshSpawner.cs

```diff
@@ -0,0 +1,86 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.AI;
+
+public class NavMeshSpawner : MonoBehaviour
+{
+    private Vector3 pointOnNavMesh;
+    public GameObject monsterToSpawn;
+    public float countDown;  // add GameObject.Find("Spawner").GetComponent<NavMeshSpawner>().countDown = 0; on enemy script when destroyed
+    public bool doneSpawning;
+    public List<GameObject> monsters = new List<GameObject>();
+    public int spawned; // add GameObject.Find("Spawner").GetComponent<NavMeshSpawner>().spawned--; on enemy script when destroyed
+    public int amountToSpawn;
+    public int reSpawnTime;
+    public float range = 10f;
+
+    public void Update()
+    {
+        countDown += Time.deltaTime;
+
+        if (doneSpawning == false)
+        {
+            Invoke("Spawn", 0.1f);

```

Comment:
You still didn't find an alternative to using Invoke. Invoke is kind of messy and frankly not necessary here, so try and find another way to handle this.